addons/nuxeo-retention/nuxeo-retention-core/src/main/java/org/nuxeo/retention/operations/RetainDocument.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Guillaume RENARD
+ */
+package org.nuxeo.retention.operations;
+
+import java.util.Calendar;
+
+import org.nuxeo.ecm.automation.core.Constants;
+import org.nuxeo.ecm.automation.core.annotations.Context;
+import org.nuxeo.ecm.automation.core.annotations.Operation;
+import org.nuxeo.ecm.automation.core.annotations.OperationMethod;
+import org.nuxeo.ecm.automation.core.annotations.Param;
+import org.nuxeo.ecm.automation.core.collectors.DocumentModelCollector;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentRef;
+
+/**
+ * @since 11.1
+ */
+@Operation(id = RetainDocument.ID, category = Constants.CAT_DOCUMENT, label = "Retain", description = "Turn the input document into a record and retain it until the until date. Returns back the retained document.")
+public class RetainDocument {
+
+    public static final String ID = "Document.Retain";
+
+    @Context
+    protected CoreSession session;
+
+    @Param(name = "until", required = false, description = "If empty, the input document will be retained indeterminately")
+    protected Calendar until;
+
+    @OperationMethod(collector = DocumentModelCollector.class)
+    public DocumentModel run(DocumentRef doc) {
+        session.makeRecord(doc);
+        session.setRetainUntil(doc, until != null ? until : CoreSession.RETAIN_UNTIL_INDETERMINATE, null);
```

Comment:
i am wondering why we set the `RETAIN_UNTIL_INDETERMINATE` when the provided value is null, because the implementation [here](https://github.com/nuxeo/nuxeo/blob/feature-NXP-27692-retention-web-ui-10.10/nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java#L2078) lets the `null` value

Reply:
That's part of the contract for this parameter: `If empty, the input document will be retained indeterminately`